controllers/healthcheck_controller.go

```diff
@@ -138,25 +139,38 @@ func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl
 	if hcSpec.Workflow.Resource != nil {
 		wfNamePrefix := hcSpec.Workflow.GenerateName
 		wfNamespace := hcSpec.Workflow.Resource.Namespace
-		repeatAfterSec := hcSpec.RepeatAfterSec
 		level := hcSpec.Level
 		sa := hcSpec.Workflow.Resource.ServiceAccount
 		now := metav1.Time{Time: time.Now()}
 		var finishedAtTime int64
 		if healthCheck.Status.FinishedAt != nil {
 			finishedAtTime = healthCheck.Status.FinishedAt.Time.Unix()
 		}
-		// workflows can be paused by setting repeatAfterSec to <= 0.
-		if repeatAfterSec <= 0 {
-			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", repeatAfterSec)
+
+		// workflows can be paused by setting repeatAfterSec to <= 0 and not specifying the schedule for cron.
+		if hcSpec.RepeatAfterSec <= 0 && hcSpec.Scheduler.Cron == "" {
+			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", hcSpec.RepeatAfterSec)
 			healthCheck.Status.Status = "Stopped"
-			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped due to spec.repeatAfterSec set to %d", repeatAfterSec)
+			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped; either spec.RepeatAfterSec or spec.Scheduler must be provided. spec.RepeatAfterSec set to %d. spec.Scheduler set to %v+", hcSpec.RepeatAfterSec, hcSpec.Scheduler)
```

Comment:
oops, I suggested the incorrect interpolation formatting previously. should be `%+v` rather than the `%v+` I had mentioned. https://golang.org/pkg/fmt/
```suggestion
			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped; either spec.RepeatAfterSec or spec.Scheduler must be provided. spec.RepeatAfterSec set to %d. spec.Scheduler set to %+v", hcSpec.RepeatAfterSec, hcSpec.Scheduler)
```